### DISCUSSION TITLE
Bitbucket hosted tap isn't cloned correctly in brew

### DISCUSSION THREAD

**Discussion by raul1991** [Homebrew/discussions#3430] [Tap maintenance and Homebrew development] [answered]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
I don't think this issue is eligible for gist logs as I haven't yet installed the formula. Instead, my issue occurs right when I run the tap command with my bitbucket repo URL.
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I am trying to add a tap which is hosted on a private bitbucket repository.

Here is what I use

`brew tap user/lazycow git@bitbucket.org:<user>/lazycow.git`

### What happened (include all command output)?

I see the logs like this

```bash
==> Tapping <user>/lazy-cow
Cloning into '/usr/local/Homebrew/Library/Taps/<user>/homebrew-lazy-cow'...
```

Here is rb file named as lazy-cow.rb (is that why its cloning into lazy-cow and not lazycow?

```ruby

class LazyCow < Formula
  desc "Install custom dev setups with one command"
  homepage ""
  url "https://statlas.prod.atl-paas.net/lazycow/lazy-cow-0.0.22-bin.tar.gz"
  mirror ""
  sha256 ""
  license ""
  version "0.0.22"
  depends_on "figlet"
  depends_on "cowsay"
  depends_on "lolcat"
  depends_on "boxes"
  depends_on "jq"

  def install
    # Fix the permissions on the global settings file.
    libexec.install Dir["*"]
    # Leave conf file in libexec. The mvn symlink will be resolved and the conf
    # file will be found relative to it
    Pathname.glob("#{libexec}/bin/*") do |file|
      next if file.directory?
      basename = file.basename
      bin.write_exec_script file, (libexec/"bin/")
    end
    puts `cowsay "All done. Going back to my farm now."`
  end
  
  def test
    assert_equal "Hello World!", shell_output("#{bin}/lazycow test").strip
  end
end
```

When I run the `--force-auto-update`, this happens
```
==> <user>/lazycow: changed remote from git@bitbucket.org:<user>/lazycow to https://github.com/<user>/homebrew-lazycow
```

### What did you expect to happen?

I expected that the clone would be done to `/Taps/<user>/lazycow` directory instead of going into a homebrew- prefixed directory. 

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
- host a tap on a bitbucket repo
- add a tap using commands given above.
- install that tap
```

**Comment by SMillerDev** [comment]
> I expected that the clone would be done to `/Taps/<user>/lazycow` directory instead of going into a homebrew- prefixed directory.

Why would you expect this? Isn't the goal to use this tap in Homebrew? In that case it needs to use the file format in brew.

**Reply by raul1991** [reply]
@SMillerDev As per the brew doc it says when `brew tap` is run without URL it is assumed its a github repo in which case the clone goes to homebrew-<tap>. However, it does not say so for the non github taps. I am really confused here.

**Reply by SMillerDev** [reply]
It says that about the URL, not about the way it stores it locally though.

**Reply by raul1991** [reply]
Yes you are absolutely right. So, you mean to say what it doesn't say shouldn't be assumed and what has happened is correct?

So that means, all repos irrespective of where they are hosted go to homebrew-<>?

**Reply by SMillerDev** [reply]
I'd make it more definitive than that. Any files a brew command except `brew install` downloads should be considered a private brew api and could change without notice.

Today the answer is AFAIK yes, in a day it might be no.

**Reply by raul1991** [reply]
ok. Can you please explain what happened wrong when I ran the --force-update ? Why does it complain about the remote being moved?

**Comment by carlocab** [comment]
It's hard to say anything about this since this isn't possible to reproduce. In particular, the command you say you have issues with
```
brew tap user/lazycow git@bitbucket.org:<user>/lazycow.git
```
isn't in a form that can be run.

**Reply by raul1991** [reply]
@carlocab Logically, this command doesn't have the issue. I am just not sure it did the right thing and hence I wanted some clarification around it.

Can you jump in with me and @SMillerDev  thread above ?

**Comment by raul1991** [comment, marked as answer]
In case someone has the same error, here is what I was doing wrong.

My repository wasn't prefixed with `homebrew` hence brew assumed it to be hosted on github and set the wrong origin for that cloned repo.

So to fix it I changed my repo's name to homebrew-<repo>
